google-cloud-storage/google-cloud-storage.gemspec

```diff
@@ -21,7 +21,7 @@ Gem::Specification.new do |gem|
   gem.add_dependency "google-cloud-core", "~> 1.6"
   gem.add_dependency "google-apis-core", "~> 0.13"
   gem.add_dependency "google-apis-iamcredentials_v1", "~> 0.18"
-  gem.add_dependency "google-apis-storage_v1", "~> 0.38"
+  gem.add_dependency "google-apis-storage_v1", "~> 0.40"
```

Comment:
The `soft_deleted` field on get_bucket was introduced in storage_v1 0.42 version. [Ref](https://github.com/googleapis/google-api-ruby-client/commit/2ae24579a2e0e694e66090103ba88d904811a1d9#diff-7c3ec9c02ad6010c2dd866c170f4575f9d687553568f5c3bf5d15fe7167b5a57)
Which version were you using in local?
Also generally, just update to the latest version of the dependency when you do this. 
This should solve the issue.